Sources/Features/Network/BacktraceApi.swift

```diff
@@ -56,3 +61,42 @@ extension BacktraceApi: BacktraceApiProtocol {
         }
     }
 }
+
+extension BacktraceApi: BacktraceMetricsApiProtocol {
+
+    func sendMetrics(_ payload: SummedEventsPayload, url: URL) throws -> BacktraceMetricsResult {
+        try sendMetrics(payload, url: url, metricsDelegate: summedEventsDelegate)
+    }
+
+    func sendMetrics(_ payload: UniqueEventsPayload, url: URL) throws -> BacktraceMetricsResult {
+        try sendMetrics(payload, url: url, metricsDelegate: uniqueEventsDelegate)
+    }
+
+    func sendMetrics<T: Event>(_ payload: Payload<T>, url: URL, metricsDelegate: BacktraceMetricsDelegate?)
+        throws -> BacktraceMetricsResult {
+        let payload = payload
+
+        do {
+            // create request
+            var urlRequest = try MetricsRequest(url: url, payload: payload).request
+
+            // modify request before sending
+            urlRequest = metricsDelegate?.willSendRequest?(urlRequest) ?? urlRequest
+            BacktraceLogger.debug("Will send URL request to metrics API: \(urlRequest)")
+
+            // send request
+            let httpResponse = try networkClient.sendMetrics(request: urlRequest)
+
+            // get result
+            BacktraceLogger.debug("Received HTTP response from metrics API: \(httpResponse)")
+            let result = httpResponse.result()
+            metricsDelegate?.serverDidRespond?(result)
+
+            return result
+        } catch {
+            BacktraceLogger.error("Connection for \(payload) failed with error: \(error)")
+            metricsDelegate?.connectionDidFail?(error)
```

Comment:
Shouldn't we try a retry here? (exponential backoff etc)

Reply:
We are not doing retry for V1 of iOS metrics, @vlussenburg